ndk/src/font.rs

```diff
@@ -0,0 +1,401 @@
+    pub const EXTRA_BOLD: FontWeight = FontWeight(ffi::AFONT_WEIGHT_EXTRA_BOLD as u16);
+    pub const BLACK: FontWeight = FontWeight(ffi::AFONT_WEIGHT_BLACK as u16);
+    pub const MAX: FontWeight = FontWeight(ffi::AFONT_WEIGHT_MAX as u16);
+}
+
+impl fmt::Display for FontWeight {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match *self {
+            FontWeight::THIN => writeln!(f, "Thin"),
+            FontWeight::EXTRA_LIGHT => writeln!(f, "Extra Light (Ultra Light)"),
+            FontWeight::LIGHT => writeln!(f, "Light"),
+            FontWeight::NORMAL => writeln!(f, "Normal (Regular)"),
+            FontWeight::MEDIUM => writeln!(f, "Medium"),
+            FontWeight::SEMI_BOLD => writeln!(f, "Semi Bold (Demi Bold)"),
+            FontWeight::BOLD => writeln!(f, "Bold"),
+            FontWeight::EXTRA_BOLD => writeln!(f, "Extra Bold (Ultra Bold)"),
+            FontWeight::BLACK => writeln!(f, "Black (Heavy)"),
+            _ => writeln!(f, "{}", self.0),
+        }
+    }
+}
+
+/// The error type returned when an invalie font weight value is passed.
+#[derive(Debug)]
+pub struct TryFromU16Error(());
+
+impl fmt::Display for TryFromU16Error {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            fmt,
+            "font weight must be positive and less than or equal to 1000"
+        )
+    }
+}
+
+impl std::error::Error for TryFromU16Error {}
+
+impl TryFrom<u16> for FontWeight {
+    type Error = TryFromU16Error;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        FontWeight::new(value)
+    }
+}
+
+/// A 4-byte integer representing an OpenType axis tag.
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct AxisTag(u32);
+
+impl AxisTag {
+    pub const fn from_be(value: u32) -> std::result::Result<Self, TryFromU32Error> {
+        // Each byte in a tag must be in the range 0x20 to 0x7E.
+        // See https://learn.microsoft.com/en-us/typography/opentype/spec/otff#data-types for details.
+        macro_rules! check_in_valid_range {
+            ($($byte: expr),+) => {
+                $(
+                    if !(0x20 <= ($byte) && ($byte) <= 0x7E) {
+                        return Err(TryFromU32Error(()));
+                    }
+                )+
+            };
+        }
+
+        let bytes = value.to_be_bytes();
+        check_in_valid_range!(bytes[0], bytes[1], bytes[2], bytes[3]);
+
+        Ok(Self(value))
+    }
+}
+
+impl fmt::Display for AxisTag {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let bytes = self.0.to_be_bytes();
+        write!(
+            f,
+            "{}{}{}{}",
+            bytes[0] as char, bytes[1] as char, bytes[2] as char, bytes[3] as char
+        )
+    }
+}
+
+impl fmt::Debug for AxisTag {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "AxisTag(")?;
+        fmt::Display::fmt(self, f)?;
+        write!(f, " {:#x})", self.0)
```

Comment:
What is wrong with using:
```suggestion
        write!(f, "AxisTag({} {:x})", self, self.0)
```

Reply:
No problem, even I don't understand why I wrote so 🤣 . BTW, Do you think Prefixing `0x` is unnecessary in this case?

Reply:
Fixed! I'll keep the prefix `0x`.
```rust
write!(f, "AxisTag({} {:#x})", self, self.0)
```

Reply:
Indeed, I didn't mean to drop the `#`, thanks!